commit 0e7a94f9f71945bceab37eb98bf581c2d973b2f1 (refs/changes/32/132/1)
Author: Dave Pacheco <dap@joyent.com>
Date:   2016-07-22T10:52:53-07:00 (3 years, 3 months ago)
    
    MANTA-2950 postgres transform job buffers far too much in memory

diff --git a/lib/sqltojson.js b/lib/sqltojson.js
index 9407c48..fc51896 100644
--- a/lib/sqltojson.js
+++ b/lib/sqltojson.js
@@ -69,6 +69,9 @@ function SqlToJsonStream(opts) {
         assert.object(opts, 'opts');
         assert.optionalArrayOfString(opts.tables, 'tables');
         opts.decodeStrings = false;
+        if (!opts.hasOwnProperty('highWaterMark')) {
+                opts.highWaterMark = 16;
+        }
         Transform.call(self, opts);
 
         //this._writableState.objectMode = true; // read objects
diff --git a/lib/table_demux.js b/lib/table_demux.js
index 2cf9227..3fd854b 100644
--- a/lib/table_demux.js
+++ b/lib/table_demux.js
@@ -25,6 +25,7 @@ module.exports = TableDemuxStream;
 function TableDemuxStream() {
         var self = this;
         Writable.call(self, {
+                highWaterMark: 16,
                 objectMode: true // read objects
         });
         self.stream = null;
